src/agent/codecoverageenabler.ts

```diff
@@ -0,0 +1,241 @@
+            includeFilter: includeFilter,
+            excludeFilter: excludeFilter
+        };
+    }
+}
+
+export class JacocoGradleCodeCoverageEnabler extends CodeCoverageEnabler {
+    //-----------------------------------------------------
+    // Enable code coverage for Jacoco Gradle Builds
+    // - enableCodeCoverage: CodeCoverageProperties  - ccProps
+    //-----------------------------------------------------    
+    public enableCodeCoverage(ccProps: ccifm.CodeCoverageProperties): Q.Promise<boolean> {
+        var defer = Q.defer<boolean>();
+        var _this = this;
+
+        var buildFile = ccProps.buildFile;
+        var classFilter = ccProps.classFilter;
+        var isMultiModule = ccProps.isMultiModule;
+        var classFileDirs = ccProps.classFileDirs;
+        var reportDir = ccProps.reportDir;
+        var codeCoveragePluginData = null;
+
+        var filter = _this.extractFilters(classFilter);
+        var jacocoExclude = _this.applyJacocoFilterPattern(filter.excludeFilter);
+        var jacocoInclude = _this.applyJacocoFilterPattern(filter.includeFilter);
+
+        if (isMultiModule) {
+            codeCoveragePluginData = ccc.jacocoGradleMultiModuleEnable(jacocoExclude.join(","), jacocoInclude.join(","), classFileDirs, reportDir);
+        } else {
+            codeCoveragePluginData = ccc.jacocoGradleSingleModuleEnable(jacocoExclude.join(","), jacocoInclude.join(","), classFileDirs, reportDir);
+        }
+
+        if (codeCoveragePluginData) {
+            _this.executionContext.info("Code Coverage data will be appeneded to build file: " + buildFile);
+            utilities.appendTextToFileSync(buildFile, codeCoveragePluginData);
+            _this.executionContext.info("Appended code coverage data");
+
+            defer.resolve(true);
+        } else {
+            _this.executionContext.warning("Unable to append code coverage data");
+            defer.resolve(false);
+        }
+
+        return defer.promise;
+    }
+
+    private applyJacocoFilterPattern(filter: string): string[] {
+        var jacocoFilter = [];
+        var _this = this;
+
+        if (!utilities.isNullOrWhitespace(filter)) {
+            str(utilities.trimToEmptyString(filter)).replaceAll(".", "/").split(":").forEach(exFilter => {
+                if (exFilter) {
+                    jacocoFilter.push(str(exFilter).endsWith("*") ? ("'" + exFilter + "/**'") : ("'" + exFilter + ".class'"));
+                }
+            });
+        }
+
+        return jacocoFilter;
+    }
+}
+
+export class JacocoMavenCodeCoverageEnabler extends CodeCoverageEnabler {
+    //-----------------------------------------------------
+    // Enable code coverage for Jacoco Maven Builds
+    // - enableCodeCoverage: CodeCoverageProperties  - ccProps
+    //-----------------------------------------------------    
+    public enableCodeCoverage(ccProps: ccifm.CodeCoverageProperties): Q.Promise<boolean> {
+        var defer = Q.defer<boolean>();
+
+
```

Comment:
Should we log some message that this is not supported. same for other unsupported scenarios